src/pages/console/formbuilder/lifecycle.mdx

```diff
@@ -0,0 +1,163 @@
+
+<BlockSwitcher>
+
+<Block name="Form is connected to a data model">
+
+If your form **is connected** to a data model, use the `onSubmit` hook to customize the form data before they are saved to the cloud. The form data that's returned from the `onSubmit` handler will be saved to the cloud.
+
+For example, if you want to trim all the string data before saving it:
+
+```jsx
+<HomeCreateForm
+    onSubmit={(fields) => {
+        // Example function to trim all string inputs
+        const updatedFields = {}
+        Object.keys(fields).forEach(key => {
+            if (typeof fields[key] === 'string') {
+                updatedFields[key] = fields[key].trim()
+            } else {
+                updatedFields[key] = fields[key]
+            }
+        })
+        return updatedFields
+    }}
+/>
+```
+
+</Block>
+
+<Block name="Form is not connected to a data model">
+
+If your form **is not connected** to a data model, use the `onSubmit` hook to submit the form data and dismiss or reroute the form. 
+
+For example, let's assume you have an API `/submit-home-data` and want to hide the form after submission:
+
+```jsx
+import { useState } from 'react'
+import { HomeCreateForm } from './ui-components'
+
+function App() {
+  const [showForm, setShowForm] = useState(true)
+
+  return (
+    {showForm &&
+      <HomeCreateForm onSubmit={async (fields) => {
+      const response = await fetch("/submit-home-data", {
+        method: "POST",
+        headers: {
+          "Accept": "application/json",
+          "Content-Type": "application/json"
+        },
+        body: JSON.stringify(fields) // Pass in form data
+      })
+
+      if (response.status === 200) {
+        setShowForm(false) // Hide the form
+      }
+    }}/>}
+  )
+}
+```
+
+</Block>
+
+</BlockSwitcher>
+
+### Handle form data successfully saving to the cloud - onSuccess
+
+If your form **is connected** to a data model, use the `onSuccess` hook to dismiss form after the form data has been successfully submitted. 
+
+```jsx
+import { useState } from 'react'
+import { HomeCreateForm } from './ui-components'
+
+function App() {
+  const [showForm, setShowForm] = useState(true)
+
+  return (
+    {showForm &&
+      <HomeCreateForm onSuccess={() => {
+        setShowForm(false) // Hide the form
+      }}/>}
+  )
+}
+```
+
+### Handle form submission errors - onError
+
+If your form is connected to a data model, you might encounter additional errors during the submit process. You can log these errors and present an alert to customers by using the `onError` handler.
+
+```jsx
+import { HomeCreateForm } from './ui-components'
+
+function App() {
+  return (
+    <HomeCreateForm onError={(error) => {
+      console.log(error)
+    }}/>
+  )
+}
+```
+
+## Handle user clicking on **Cancel** action button - onCancel
+
+If the user clicks on the **Cancel** action button, you can use the `onCancel` event to hide the form or reroute the customer to another page.
```

Comment:
`form or reroute`

I think reroute implies they were already being routed so this should probably just be `form or route`